### PR TITLE
Migrate EcalMonitorPrescaler deprecated legacy EDFilter

### PR DESCRIPTION
**Pull request by alejands**
#### PR description:

Migrate EcalMonitorPrescaler to `edm::global::EDFilter<edm::RunCache<...>>` as part of https://github.com/cms-sw/cmssw/issues/36404

The event counters used for prescaling are now stored in a `edm::RunCache<PrescaleCounter>`, where `PrescaleCounter` is a struct wrapper for an array of `std::atomic<unsigned>`

For more details on  `edm::global::EDFilter<edm::RunCache<...>>` (and future reference): https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkGlobalModuleInterface#edm_RunCacheT

#### PR validation:

CMSSW compiles with no `CMS_DEPRECATED` warnings

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by emanueleusai** [comment]
please test

**Comment by alejands** [comment]
From what I can see, it appears that the failed test is not related to this PR or this module.

**Comment by emanueleusai** [comment]
indeed, let's retest

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
type ecal

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
I imagine that the Tracker DQM differences reported for wf 11634.7 are totally unrelated: correct?

File: DQM/EcalCommon/interface/EcalMonitorPrescaler.h [hunk 1]
```diff
@@ -2,26 +2,38 @@
...
-  bool filter(edm::Event &, edm::EventSetup const &) override;
+  std::shared_ptr<ecaldqm::PrescaleCounter> globalBeginRun(edm::Run const &, edm::EventSetup const &) const override;
+  bool filter(edm::StreamID, edm::Event &, edm::EventSetup const &) const override;
+  void globalEndRun(edm::Run const &, edm::EventSetup const &) const override;
 
 private:
-  enum Prescalers { kPhysics, kCosmics, kCalibration, kLaser, kLed, kTestPulse, kPedestal, nPrescalers };
-
-  static uint32_t filterBits_[nPrescalers];
+  static uint32_t filterBits_[ecaldqm::nPrescalers];
```
Comment (perrotta) [inline DQM/EcalCommon/interface/EcalMonitorPrescaler.h:32]: Should this be `const` (see also the static analyzer report)?

Comment (alejands) [reply DQM/EcalCommon/interface/EcalMonitorPrescaler.h:32]: I believe so. There's no reason for this *not* to be `static const`. I'll update the PR and the backports

**Comment by alejands** [comment]
> I imagine that the Tracker DQM differences reported for wf 11634.7 are totally unrelated: correct?

@perrotta As far as I know, these tracker plots are totally unrelated. From what I can see, this module is not used outside of ECAL DQM contexts.

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**